bot.py

```diff
@@ -87,116 +91,166 @@ def translate_slack_usernames(self, unchecked_authors):
         """
-        check_call([in_script_dir("finish_release.sh"), self.repo_dir, self.version])
-        self.say("Merged evil scheme {}! Now deploying to production...".format(self.version))
+        check_call([in_script_dir("finish_release.sh"), self.repo_dir, version])
+        await self.say("Merged evil scheme {}! Now deploying to production...".format(version))
         check_call([in_script_dir("wait_for_deploy.sh"), self.repo_dir, self.prod_hash_url, "release"])
-        self.say(
+        await self.say(
             "My evil scheme {} has been released to production. "
-            "And by 'released', I mean completely...um...leased.".format(self.version)
+            "And by 'released', I mean completely...um...leased.".format(version)
         )
 
-    def message_if_unchecked(self):
+    async def message_if_unchecked(self, version):
         """
         Send a message next morning if any boxes are not yet checked off
         """
-        unchecked_authors = get_unchecked_authors(self.org, self.repo, self.version)
+        unchecked_authors = get_unchecked_authors(self.org, self.repo, version)
         if unchecked_authors:
             slack_usernames = self.translate_slack_usernames(unchecked_authors)
-            self.say(
+            await self.say(
                 "What an unexpected surprise! "
                 "The following authors have not yet checked off their boxes: {}".format(
                     ", ".join(slack_usernames)
                 )
             )
 
+    async def delay_message(self, version):
+        """sleep until 10am next day, then message"""
+        now = datetime.now()
+        tomorrow_at_10 = next_workday_at_10(now)
+        await asyncio.sleep((tomorrow_at_10 - now).total_seconds())
+        await self.message_if_unchecked(version)
+
+    async def handle_message(self, content, loop):
+        """handle the message"""
+        if has_command(['release'], content):
+            try:
+                version = get_version_number(content)
+            except Exception as exception:
+                await self.say("{}".format(exception))
+            else:
+                if has_command(['start', 'release'], content):
+                    loop.create_task(self.delay_message(version))
+                    await self.do_release(version)
+                    await self.wait_for_checkboxes(version)
+                elif has_command(['finish', 'release'], content):
+                    await self.finish_release(version)
+                else:
+                    await self.say("Oooopps! Invalid command format")
+        else:
+            await self.say("Want to start a release? hhhmmmmm")
+
+
+def get_version_number(content):
+    """return version number at the end of the message"""
+    pattern = re.compile('^[0-9.]+$')
+    if pattern.match(content[-1]):
+        return content[-1]
+    else:
+        raise Exception("Invalid version number")
+
+
+def has_command(words, content):
```

Comment:
This should be a prefix check. If words is `['start', 'release']` we should only match `content` if it looks like `start release 123`, rather than `start a release` or `release start` or anything else like that. The more specific we are with the input we accept, the easier it will be to identify and fix bugs